improvement(marginfi-v2-ui): add Coinbase and SquadsX wallets to extended options

masterzorgon opened the pull request:
## Issue
Both the Coinbase and SquadsX team raised issues with me this past week that users are unable to use their respective wallets to onboard to marginfi. 

## Solution
I add SquadsX and Coinbase wallet adapters inside the `src/hooks/useAvailableWallet.ts` file and conditionally added these adapters to the `addAbsentWallets` function.

k0beLeenders reviewed (changes requested):
Can you rebase this branch on main @masterzorgon

masterzorgon commented:
> Can you rebase this branch on main @masterzorgon

rebased!

k0beLeenders reviewed (changes requested):
I see a few unrelated changes, something seems to have gone wrong with the rebase

k0beLeenders commented on apps/marginfi-v2-ui/src/utils/mrgnActions.ts:
```diff
@@ -879,6 +879,7 @@ export async function mintLstToken({
 // Helpers //
 // ------------------------------------------------------------------//
 
+// PDA not being able to create a bank?
```
what do you mean with this?

masterzorgon replied:
this was a comment i made while debugging a SquadsX issue. it was a note to myself and forgot to delete it. i was mistaken in thinking at our code wasn't allowing off-curve signers to create banks

k0beLeenders commented:
I made some changes to the onboarding process. Previously, we filtered the wallet list to show only relevant options for social, Sol, and Eth flows. However, this caused an issue where users could not find their wallet during onboarding. To address this, I removed the forced filter for the Sol and Eth flows so that all installed wallets are now visible. Additionally, I have reordered the wallets to prioritize certain ones. 

let me know your thoughts @masterzorgon @chambaz

chambaz commented on apps/marginfi-v2-ui/src/components/common/Wallet/components/AuthenticationV2/AuthDialog.tsx:
```diff
@@ -43,7 +43,7 @@ export const AuthDialog = () => {
   }, []);
 
-  const [flow, setFlow] = React.useState<AuthFlowType>(mainFlow);
+  const [flow, setFlow] = React.useState<AuthFlowType>("ONBOARD_ETH");
```
woah, close one. lets make sure to fully test everything :)

chambaz commented:
> I made some changes to the onboarding process. Previously, we filtered the wallet list to show only relevant options for social, Sol, and Eth flows. However, this caused an issue where users could not find their wallet during onboarding. To address this, I removed the forced filter for the Sol and Eth flows so that all installed wallets are now visible. Additionally, I have reordered the wallets to prioritize certain ones.
> 
> let me know your thoughts @masterzorgon @chambaz

I tend to agree here. I think the way we had attempted to do it makes sense, reducing choice paralysis, and simplifying the flow. however from feedback users and wallet providers expect to see their option available and I think your latest solution is the easiest option. one day I pray for a true Wallet Standard and this implementation nightmare goes away.

masterzorgon commented:
> I made some changes to the onboarding process. Previously, we filtered the wallet list to show only relevant options for social, Sol, and Eth flows. However, this caused an issue where users could not find their wallet during onboarding. To address this, I removed the forced filter for the Sol and Eth flows so that all installed wallets are now visible. Additionally, I have reordered the wallets to prioritize certain ones.
> 
> let me know your thoughts @masterzorgon @chambaz

masterzorgon commented:
> I made some changes to the onboarding process. Previously, we filtered the wallet list to show only relevant options for social, Sol, and Eth flows. However, this caused an issue where users could not find their wallet during onboarding. To address this, I removed the forced filter for the Sol and Eth flows so that all installed wallets are now visible. Additionally, I have reordered the wallets to prioritize certain ones.
> 
> let me know your thoughts @masterzorgon @chambaz



> I made some changes to the onboarding process. Previously, we filtered the wallet list to show only relevant options for social, Sol, and Eth flows. However, this caused an issue where users could not find their wallet during onboarding. To address this, I removed the forced filter for the Sol and Eth flows so that all installed wallets are now visible. Additionally, I have reordered the wallets to prioritize certain ones.
> 
> let me know your thoughts @masterzorgon @chambaz

I see that the SquadsX Wallet option is not available during the onboarding flow.

<img width="640" alt="Screenshot 2024-07-05 at 3 23 12 PM" src="https://github.com/mrgnlabs/mrgn-ts/assets/155211932/23617a0d-42f6-494e-be0a-53ecb57e73ab">

k0beLeenders commented:
Rebasing on main rn, but should be good to go when @chambaz approves it
